RetailCoder.VBE/Inspections/VariableRequiresSetAssignmentEvaluator.cs

```diff
@@ -0,0 +1,160 @@
+﻿using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rubberduck.Inspections
+{
+    class VariableRequiresSetAssignmentEvaluator
+    {
+        private readonly RubberduckParserState _parserState;
+        public VariableRequiresSetAssignmentEvaluator(RubberduckParserState parserState)
+        {
+            _parserState = parserState;
+        }
+
+        public IEnumerable<Declaration> GetDeclarationsPotentiallyRequiringSetAssignment()
+        {
+            var interestingDeclarations = _parserState.AllUserDeclarations.Where(item =>
+                    IsVariableOrParameter(item)
+                    && !item.IsSelfAssigned
+                    && TypeIsAnObjectOrVariant(item));
+
+            var interestingMembers = _parserState.AllUserDeclarations.Where(item =>
+                    IsMemberWithReturnType(item)
+                    && item.IsTypeSpecified
+                    && TypeIsAnObjectOrVariant(item));
+
```

Comment:
These two should be combined - you're iterating `AllUserDeclarations` twice, doing *almost* the exact same work twice.